lnrpc/invoicerpc/invoice.proto

```diff
@@ -144,43 +226,71 @@ message Invoice {
     bool private = 15;
 
     /**
-    The "add" index of this invoice. Each newly created invoice will increment
-    this index making it monotonically increasing. Callers to the
-    SubscribeInvoices call can use this to instantly get notified of all added
-    invoices with an add_index greater than this one.
+    The number of this invoice. Each newly created invoice will increment
+    this number making it monotonically increasing. 
     */
-    uint64 add_index = 16;
+    uint64 invoice_number = 16;
 
     /**
-    The "settle" index of this invoice. Each newly settled invoice will
-    increment this index making it monotonically increasing. Callers to the
-    SubscribeInvoices call can use this to instantly get notified of all
-    settled invoices with an settle_index greater than this one.
-    */
-    uint64 settle_index = 17;
+    The "last update" index of this invoice. last_update_index is a 
+    monotonically increasing value that assigns an order to all invoice updates.
+    It can be used by clients to catch up with missed updates. 
 
-    /// Deprecated, use amt_paid_sat or amt_paid_msat.
-    int64 amt_paid = 18;
-
-    /**
-    The amount that was accepted for this invoice, in satoshis. This will ONLY
-    be set if this invoice has been settled. We provide this field as if the
-    invoice was created with a zero value, then we need to record what amount
-    was ultimately accepted. Additionally, it's possible that the sender paid
-    MORE that was specified in the original invoice. So we'll record that here
-    as well.
+    Whenever a change to any invoice in the system is made, last_update_index
+    is incremented and stored in the updated invoice. 
     */
-    int64 amt_paid_sat = 19;
```

Comment:
Redundant and deprecated fields removed.

Reply:
Why?

Reply:
Today I needed to do some serious investigation only to find out that the cause of a problem was a rounding error from msat to sat.

Reply:
Precisely because rounding can be easy to mess up. Over REST, these will be serialized as strings and require custom parsing and arithmetic. I'd rather have rounding logic live inside LND, and allow clients to choose what granularity they choose to display.

In your case it sounds like the API should accept msat instead of (or in addition to) sat?

Reply:
Yes, in the case that I had, the API accepting msat would be better. And how about reporting all fields in sat and msat (to prevent client rounding), but accepting only in msat? There is not much that can go wrong with converting from sat to msat. But then as you said, there is still the serialization issue. Okay, let's leave it in.